### PR TITLE
Added support for outlier and field-based monitors

### PR DESCRIPTION
**Pull request by kumar-avijit**
This adds support for 6 new monitor condition types:
* LogsStaticCondition
* MetricsStaticCondition
* LogsOutlierCondition
* MetricsOutlierCondition
* LogsMissingDataCondition
* MetricsMissingDataCondition

(@rsanbhad , @tarunk2 : I am still to fix some tests, but I want to get your early opinion on how justified the loss of strict schema checking for the sake of backwards compatibility is here.)

### CODE REVIEW

File: website/docs/r/monitor.html.markdown [hunk 1]
```diff
@@ -235,6 +235,88 @@ Additional data provided in state:
   - `Normal`
   - `Disabled`
 
+### Triggers arguments
```
Comment (rsanbhad) [inline website/docs/r/monitor.html.markdown:238]: Thanks for adding these 🙏  🙇

**Review by rsanbhad — APPROVED** [review]
lgtm

**Comment by rsanbhad** [comment]
@kumar-avijit I think you'll need to enable all the backend FF for the org that we use for terraform acceptance tests on us2 - @seansain do you know the orgId for that org?

**Comment by kumar-avijit** [comment]
Thanks @rsanbhad for the review. @seansain reached out to me with the org ID: `00000000009A0DAB`.

**Comment by rsanbhad** [comment]
I added it to knobs PR https://github.com/Sanyaku/knobs/pull/21778 since I had to enable for a new customer anyway - you can re-trigger PR builder after that knobs PR is merged and takes effect

**Comment by kumar-avijit** [comment]
superseded by https://github.com/SumoLogic/terraform-provider-sumologic/pull/239

**Pull request closed without merging**